website/docs/overloads.md

```diff
@@ -0,0 +1,219 @@
+  the call site.
+- The argument names can be unique in each definition as well. In this example:
+  `id` vs `ids`.
+- Each method can be documented independently. Documentation above a method
+  definition is surfaced when hovering over a method call and when selecting
+  completion items.
+- Since the method name controls which method is selected, untyped arguments do
+  not interfere with the inferred return type of a method call.
+
+### Multiple methods, but sharing a common implementation
+
+The biggest downside of this approach deals with sharing code.
+
+In most cases, possible to implement one methods in terms of the others. For
+example, with our `find_one`/`find_many` example above:
+
+```ruby
+sig { params(id: String).returns(MyModel) }
+def find_one(id)
+  result = find_many([id])
+  raise "find_many did not return a single result" unless result.size == 1
+  result.fetch(0)
+end
+```
+
+In cases when this is not possible, another option is to do something like this:
+
+```ruby
+sig do
+  params(what: T.any(String, T::Array[String]))
+    .returns(T.any(MyModel, T::Array[MyModel]))
+end
+private def _find_impl
+  # ...
+end
+
+sig { params(id: String).returns(MyModel) }
+def find_one(id)
+  T.unsafe(_find_impl(id))
+end
+
+sig { params(ids: T::Array[String]).returns(T::Array[MyModel]) }
+def find_many(ids)
+  T.unsafe(_find_impl(ids))
+end
+```
+
+In this example, we define the common logic in a private `_find_impl` method
+with a signature that accepts the superset of all arguments, and returns a
+superset of all return types. We then call that method inside public methods
+with more specific types, using `T.unsafe`. This keeps usage of `T.untyped`
+internal to the class's private implementation while exposing a typed public
+API.
+
+There are a couple other ways to accomplish a similar effect:
+
+- Mark the return type of `_find_impl` as `T.untyped`, to avoid needing a
+  `T.unsafe` at each call site. Or omit the signature on `_find_impl` entirely.
+
+- Define the signatures for `find_one` and `find_many` in an [RBI file](rbi.md)
+  alongside the source file, and use something like
+
+  ```ruby
+  T.unsafe(self).alias_method(:find_one, :_find_impl)
+  T.unsafe(self).alias_method(:find_many, :_find_impl)
+  ```
+
+  to create method aliases to the `_find_impl` method, but using `T.unsafe` to
+  hide those aliases from Sorbet, so that the RBI definitions are all Sorbet
+  sees.
+
+Any of the options presented here will offer a more first-class experience than
+attempting to define a method with overloaded signatures.
+
+## Restrictions on overloaded methods
+
+Support for overloaded methods is minimal because there are restrictions on when
+they are allowed to appear and how they are allowed to be used.
+
+Overloaded signatures:
+
+1.  may only appear in RBI files, not in Ruby source files.
+
+1.  prevent the implementation of that method from being type checked.
```

Comment:
Looks like the ordering in this list is broken

Reply:
@ilyailya are you referring to the fact that they all say `1.`? Markdown ordered lists are auto-incrementing when rendered.

Reply:
You can check the rendered output to confirm this. Using `1.` for all items makes it easy to insert or remove items without needing to update all the numbers